### DISCUSSION TITLE
Content type issue when passing form data

### DISCUSSION THREAD

**Discussion by wes-kay** [quasarframework/quasar#11728] [General - Components / Directives / etc]
Trying to post my form but having issues with the body encoding  / content type. 
<img width="801" alt="Screenshot_9" src="https://user-images.githubusercontent.com/40011470/146666681-2764919b-e2e2-4e03-8efb-22a66a9be6d8.png">

```
const api = axios.create({ 
  baseURL: 'http://localhost:9990/v1',
  headers:{
    'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8'
  },  
})
```
And the page passing the form data
```
<q-form v-on:submit.prevent="onSubmit" class="q-gutter-md">
... 
export default {
  data() {
    return {
      form: {
        Title: '',
        type: '',
        description: '',
      },
    };
  },
  methods: {
    onSubmit() {
      this.$api
        .post('/create/1', this.form)
        .then((val) => {
          this.$router.push("/dashboard");
        })
        .catch((err) => console.log(err))
    },
  },

```

**Comment by MilosPaunovic** [comment]
This seems to be an issue with [another repository](https://github.com/gorilla/schema); not specifically reletad to `Quasar`, so you should post a question there.

Anyway, seems like the code which is parsing your schema is the main problem.

**Reply by wes-kay** [reply]
> This seems to be an issue with [another repository](https://github.com/gorilla/schema); not specifically reletad to `Quasar`, so you should post a question there.
> 
> Anyway, seems like the code which is parsing your schema is the main problem.

The issue is that it's auto encoding to that json format { key: {... Where is that coming from?

**Comment by MilosPaunovic** [comment]
Have you tried `this.$api.post('/create/1', { form: this.form })`?

**Comment by hawkeye64** [comment]
@wes-kay You are handling the data yourself, so your `Content-type` is not `'application/x-www-form-urlencoded;`, but rather `application/json`. If you want `'application/x-www-form-urlencoded;`, then use the form's native submit.
Reference: https://quasar.dev/vue-components/form#submitting-to-a-url-native-form-submit-

**Reply by wes-kay** [reply]
Axios and quasar looks like it's sending the data as a key value here when doing `.post('/create/1', this.form)`: 
![image](https://user-images.githubusercontent.com/40011470/146680006-e841e2bc-aed6-4958-8f13-35c9ec54aa78.png)

When that's not what I want. I need to be interpreting as form data on my API. 

I should also say I have this working in vue and Axios with the exact same settings (but I'll show you what I need using postman):
![image](https://user-images.githubusercontent.com/40011470/146679967-adbed8f5-46d5-4f2d-87d0-93e8d8680d46.png)

How can I stop quasar from sending the data as a key value here?

**Comment by hawkeye64** [comment]
@wes-kay You need to convert the JSON to `new FormData()` or use native submit, which will do all the work for you.

See here: https://stackoverflow.com/a/22783314/5359558
